process/coordinator/process.go

```diff
@@ -489,49 +489,40 @@ func (tc *transactionCoordinator) CreateMbsAndProcessTransactionsFromMe(
 		miniBlocks = append(miniBlocks, interMBs...)
 	}
 
-	tc.addTxFeeToMatchingMiniBlocks(&miniBlocks)
-
+	rewardsMBs := tc.createRewardsMiniBlocks()
+	if len(interMBs) > 0 {
+		miniBlocks = append(miniBlocks, rewardsMBs...)
+	}
+	
 	return miniBlocks
 }
 
-func (tc *transactionCoordinator) addTxFeeToMatchingMiniBlocks(miniBlocks *block.MiniBlockSlice) {
-	// add txfee transactions to matching blocks
-	interimProc := tc.getInterimProcessor(block.TxFeeBlock)
+func (tc *transactionCoordinator) createRewardsMiniBlocks() block.MiniBlockSlice {
+	// add rewards transactions to separate miniBlocks
+	interimProc := tc.getInterimProcessor(block.RewardsBlock)
 	if interimProc == nil {
-		return
+		return nil
 	}
 
-	txFeeMbs := interimProc.CreateAllInterMiniBlocks()
-	for key, mb := range txFeeMbs {
-		var matchingMBFound bool
-		for i := 0; i < len(*miniBlocks); i++ {
-			currMb := (*miniBlocks)[i]
-			if currMb.ReceiverShardID == key &&
-				currMb.SenderShardID == tc.shardCoordinator.SelfId() &&
-				currMb.Type == block.TxBlock {
-				currMb.TxHashes = append(currMb.TxHashes, mb.TxHashes...)
-				matchingMBFound = true
-				break
-			}
-		}
-
-		if !matchingMBFound {
-			mb.ReceiverShardID = key
-			mb.SenderShardID = tc.shardCoordinator.SelfId()
-			mb.Type = block.TxBlock
+	miniBlocks := make(block.MiniBlockSlice, 0)
+	rewardsMbs := interimProc.CreateAllInterMiniBlocks()
+	for key, mb := range rewardsMbs {
```

Comment:
these data should be set by the CreateAllInterMiniBlocks function. no need to do it here.

Reply:
done